### ISSUE TITLE
Migrate wiki.gpodder.org to Github

### ISSUE THREAD

**Issue by Vrihub** [gpodder/gpodder#218]
I tried to create an account on the wiki, but when I click on the "Create your account" button, I only get a blank page, and no email notification (and if I try to login with the credentials, I get a "user not found" error message).

**Comment by thp** [comment]
Yeah, we're in the slow progress of moving the Wiki from wiki.gpodder.org to here on Github: https://github.com/gpodder/gpodder/wiki -- there's no real plan or when or what, but if you want to help with moving content over, that'd be great :)

**Comment by Vrihub** [comment]
> if you want to help with moving content over, that'd be great

Why not. In the meanwhile you could add to the user manual page the link I mentioned at
https://github.com/gpodder/gpodder/issues/213#issuecomment-275905896
(assuming the old wiki pages can still be edited)

**Comment by vossad01** [comment]
It is worth mentioning that [Pandoc](pandoc.org/) can help by converting the MediaWiki markup to Markdown.

I prepared an example:  https://github.com/gpodder/gpodder/wiki/Youtube

What I did:
1. Copy MediaWiki markdown from the old Wiki (acquired by clicking edit)
1. [Online pandoc conversion][pandoc_conv]
2. RegEx Textual replace '^\`' with '  *'
3. Textual replace '\`' with ''

The above replacement probably won't apply to all files, but I think it did an okay with this one.

[pandoc_conv]: http://pandoc.org/try/?text=These+are+instructions+to+subscribe+to+user+channels+in+YouTube+-+post+June+2015.%0A%0A%0A%3D%3D+Why+do+I+need+to+do+this%3F+%3D%3D%0A%0AThere+is+no+more+transparent+relation+between+user+name+and+RSS+feed.+%0AgPodder+needs+to+use+the+google+YouTube+API+to+convert+the+user+name+(e.g.+%27%27Pentadact%27%27)+to+an+RSS+feed+(e.g.+https%3A%2F%2Fwww.youtube.com%2Ffeeds%2Fvideos.xml%3Fchannel_id%3DUCQ-twzO6v-PBgckhkrXVaDQ).%0ATo+do+so%2C+it+must+provide+an+API+key+to+google+to+be+granted+access+to+the+API.%0AThere+are+quotas+in+place+for+each+key%2C+so+the+gPodder+devs+can%27t+simply+give+a+key+for+everyone+to+use.%0AEveryone+has+to+get+his+own.%0A%0A%3D%3D+How+do+I+do+it+%3F+%3D%3D%0A%0AWe+will+follow+https%3A%2F%2Fdevelopers.google.com%2Fyoutube%2Fv3%2Fgetting-started%3Fhl%3Dfr%0A%0A1.+You+need+a+google+account+(your+gmail+address).%0A%0A2.+Go+to+https%3A%2F%2Fconsole.developers.google.com%2F%0A+++Login+with+your+google+account.%0A%0A3.+Click+the+button+%27%27Create+a+project%27%27%0A+++Name+it+whatever+you+want+(e.g.+gPodderAPI)%0A%0A4.+Go+to+%27%27API+%26+auth%27%27+%3E+%27%27APIs%27%27%0A+++Search+for+%2F+click+on+YouTube+Data+API%0A+++Click+%27%27Enable%27%27%0A%0A5.+Go+to+%27%27API+%26+auth%27%27+%3E+%27%27Credentials%27%27%0A+++In+the+%27%27Public+API+access%27%27+section%2C+click+on+the+%27%27create+new+key%27%27+button%0A+++Choose+%27%27Browser+key%27%27%0A+++Don%27t+enter+anything+in+the+HTTP+referers+text+area%0A+++Click+the+%27%27Create%27%27+button%0A%0A6.+Your+API+key+appears.%0A+++Copy+the+long+string+of+letters+and+numbers%0A%0A7.+Open+gPodder+3.8.4+or+later%0A+++Activate+the+%27%27Extras%27%27+%3E+%27%27Update+YouTube+subscriptions%27%27+menu+item%0A+++Click+YES+to+enter+your+API+key%0A+++Paste+your+API+key+in+the+%27%27Youtube+api+key+(v3)%27%27+field.%0A+++Close+the+settings+window.%0A+++Activate+the+%27%27Extras%27%27+%3E+%27%27Update+YouTube+subscriptions%27%27+menu+item+again.%0A%0A8.+You%27re+done&from=mediawiki&to=markdown_github

**Comment by Vrihub** [comment]
Ok, I gave it a try, and migrated the pages linked from the "users" section of the old wiki home page; I've linked them from the new wiki homepage.

I used pandoc to convert from mediawiki to markdown (github flavour); the conversion works quite well, only seldom requiring manual post-processing (mostly for large blocks of code, which sometimes are converted into sequences of small blocks of code, and wrongly displayed on a single line).

The only disadvantage from converting to markdown is that auto-generated TOCs from mediawiki are lost, this is relevant for the big [[User Manual]] page; for now I've added a small TOC manually, but maybe that page should be splitted into multiple smaller pages for better readability?

I think the next step would be to write a list of the remaining pages (http://wiki.gpodder.org/wiki/Special:AllPages) that should be migrated as-is, and of pages that need to be rewritten or dumped (I suspect some information in the old wiki is outdated), but I am not familiar at all with the developers documentation, so I'll leave this task to the reader(s) :)

**Comment by vossad01** [comment]
If it is later decided to shutdown wiki.gpodder.org, the hosting of the images will need to be changed since the GitHub wiki currently fetches the images from there.  This is not at all a criticism of the Vrihub's work or even necessarily a problem.  It works really well as is and it is great to have things ported.  I just wanted to mention it so it is more likely to be remembered when/if the old wiki is taken down.

**Comment by thp** [comment]
Nice work @Vrihub and @vossad01! As for the images, we could store them in the Git repository, as suggested here: https://help.github.com/articles/adding-images-to-wikis/

Interesting next steps would be:

- [ ] Splitting the "User manual" page as suggested by @Vrihub 
- [ ] Trying to create HTML documentation to ship with gPodder releases
- [ ] Clarify with @Horrendus what to do with http://wiki.gpodder.org/wiki/Libmygpo-qt 
- [ ] Check if https://github.com/gpodder/gpodder/wiki/macOS is still relevant (@elelay)
- [ ] Migrate the "Extensions" page and its subpages ("Extensions/*")
- [ ] http://wiki.gpodder.org/wiki/Translations should be migrated as "Translations"
- [ ] Clarify with @stefankoegl what should happen with the "Web Services/*" pages
- [ ] http://wiki.gpodder.org/wiki/Media_Coverage might be nice to migrate for posterity ;)
- [ ] http://wiki.gpodder.org/wiki/Release_Process should also be migrated

I've updated the wiki.gpodder.org pages to note which pages have been migrated (and where to).

**Comment by Horrendus** [comment]
Just enabled the Wiki on https://github.com/gpodder/libmygpo-qt ... seems like a good idea to migrate it there. @Vrihub and @vossad01 if you want to migrate (and maybe update) the page feel free to do that :)

**Comment by paulgit** [comment]
@vossad01 wiki.gpodder.org is not going anywhere in a hurry as it's hosted on one of my servers, however I may move it to a new server in the near future to free up resources. Only once the migration to GitHub is complete will the wiki.gpodder.org be removed.

**Comment by vossad01** [comment]
It is not exactly pretty since the GitHub wiki does not support any sort of hierarchy like on the old wiki, but I have migrated the [Extension](https://github.com/gpodder/gpodder/wiki/Extensions) pages because I went looking for information on MPRIS.

Long term, Jekyll and GitHub pages might be better than the wiki, but getting to the wiki (which is also markdown) is also a step in that direction.

**Comment by vossad01** [comment]
@paulgit Here is the converted source for your page.  I don't have permission to edit that wiki it seems.

That leaves only the Web Services pages needing determination and possible conversion according to the list given above.

```
General
-------

### General Information

libmygpo-qt is a Qt/C++ library wrapping the [gpodder.net Webservice](Web_Services "wikilink"). It is currently used in Amarok & Clementine to retrieve a directory of podcasts & to synchronize podcast subscriptions with [gpodder.net](Web_Services "wikilink")

-   License: LGPLv2.1+

### Contact Information

Current Maintainer of libmygpo-qt is **Stefan Derkits** (HorusHorrendus @ \#gpodder)

If you need any help or Information about libmygpo-qt write to the [Mailing List](Mailing_List "wikilink")

### Links

-   [Doxygen Documentation](http://stefan.derkits.at/libmygpo-doc/)
-   [Bugtracker for libmygpo-qt](https://bugs.gpodder.org/)
-   [GIT Repo](https://github.com/gpodder/libmygpo-qt)

### Build Dependencies

-   CMake 2.6.2
-   \+ all dependencies listed below

### Dependencies

-   Qt &gt;= 4.6
-   [QJson](http://qjson.sourceforge.net/)

### Qt5 support

libmygpo-qt version 1.0.8 supports also building the library with Qt5. The API is the same as in the Qt4 library. To build the library with Qt5 add "-DBUILD\_WITH\_QT4=OFF" to your CMake command line. Please note that Qt5 support is still experimental. If you find anything to not work with Qt5, please report a bug.

### Used for

-   [Amarok](http://amarok.kde.org)
-   [Clementine](http://www.clementine-player.org/)

Current Version
---------------

-   Version: 1.0.8

Tarball: <http://stefan.derkits.at/files/libmygpo-qt/libmygpo-qt.1.0.8.tar.gz>
sha256sum: 83716ea5cd6c0010d4531dd2b0c4e83c12d67b738da6aa15c932fc5901902e81
md5sum: cb67c86919171d6d2356dfb59c3b9571

Future
------

Just some quick Ideas what could be done better with libmygpo-qt 2.0 ;)

-   Code Reusabillity (e.g. Slots error, parse(QByteArray) & parseData are the same in many classes)
    -   possible in v1.X? Only Private Classes need to be changed to use a "ParsableResult" as Parent ...
-   Investigate if it is possible to return QString instead of QNetworkReply\* for Data that will not be parsed (OPML, TXT, XML)
-   Merge Functions that return unparsed data in different formats to a function with an enum parameter for the format
-   Memory vs. RunTime in Objects (e.g. DeviceSyncResult): Save Lists & Co only as QVariant (needed for Properties) and parse on demand or save parsed & QVariant (still possible in v1.X)
```

**Comment by adamvoss** [comment]
Most of the tasks here have been completed.  I have created new issues for the remaining tasks and am closing this in favor of gpodder/gpodder.github.io#9 and gpodder/gpodder.github.io#10.
